File: nautilus-server/src/main/java/co/kaioru/nautilus/server/IShard.java
package co.kaioru.nautilus.server;

import co.kaioru.nautilus.core.config.IConfigurableRemote;
import co.kaioru.nautilus.server.config.ShardConfig;

import java.rmi.RemoteException;
import java.util.Collection;

public interface IShard<C extends ICluster, CO extends ShardConfig> extends IConfigurableRemote<CO>, IDaemon<CO> {

	Collection<? extends C> getClusters() throws RemoteException;

	void registerCluster(C cluster) throws RemoteException;

	void deregisterCluster(C cluster) throws RemoteException;

}

File: nautilus-server/src/main/kotlin/co/kaioru/nautilus/interop/IShard.kt
package co.kaioru.nautilus.interop

import co.kaioru.nautilus.interop.info.ShardInfo
import java.rmi.RemoteException

interface IShard<out I : ShardInfo, C : ICluster<*>> : INode<I> {
	val clusters: MutableCollection<C>
		@Throws(RemoteException::class) get

	@Throws(RemoteException::class)
	fun registerCluster(cluster: C)

	@Throws(RemoteException::class)
	fun deregisterCluster(cluster: C)
}
